### PR TITLE
Fix error handling bug

### LINKED ISSUE #20896: Error handling while generating a JWT
**Description:**

jwt:issueJWT supposed to return string|error as per the document https://ballerina.io/learn/api-docs/ballerina/jwt/functions.html#issueJwt. However it does not properly return an error. Instead it throws an error internally and exit. 

**Steps to reproduce:**
An example:
```
public function generateAPIKey() {
       crypto:KeyStore keyStore = {
            path: path,
            password: password
        };

        jwt:JwtKeyStoreConfig config = {
            keyStore: keyStore,
            keyAlias: wrong alias,
            keyPassword: password
        };
       ...
       ...
        printDebug(API_KEY_UTIL, "API Key is being issued.. .");
        string | error apiKey = jwt:issueJwt(header, jwtPayload, config);
        printDebug(API_KEY_UTIL, "API Key issuing process completed");
}
```
 if we provide an invalid alias it will exit with 
```
2020-02-06 17:15:13,212 DEBUG [APIKeyUtil] [-] API Key is being issued.. .
error: {ballerina/crypto}Error message=Unable to open keystore: keystore password was incorrect
	at ballerina:externDecodePrivateKey(crypto.bal:341)
	   ballerina:decodePrivateKey(crypto.bal:338)
	   ballerina.jwt:issueJwt(jwt_issuer.bal:70)
	   wso2.gateway.src.gateway.utils.api_key_utils:generateAPIKey(/src/gateway/utils/api_key_utils.bal:98)
	   wso2.pet.apiKeyService__service_4:apiKeyResource(tokenServices.bal:144)
```

Similarly, If we provide a wrong password as above, it will exit with
```
2020-02-06 17:15:13,212 DEBUG [APIKeyUtil] [-] API Key is being issued.. . 
error: java.lang.NullPointerException 
	at ballerina:externDecodePrivateKey(crypto.bal:65535)
	   ballerina:decodePrivateKey(crypto.bal:338)
	   ballerina.jwt:issueJwt(jwt_issuer.bal:70)
	   wso2.gateway.src.gateway.utils.api_key_utils:generateAPIKey(/src/gateway/utils/api_key_utils.bal:98)
	   wso2.pet.apiKeyService__service_4:apiKeyResource(tokenServices.bal:144)
```
Note that printDebug(API_KEY_UTIL, "API Key is being issued.. ."); printed while printDebug(API_KEY_UTIL, "API Key issuing process completed"); did not.

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Fix error handling bug while generating a JWT

Fixes #20896

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java [hunk 1]
```diff
@@ -85,6 +78,13 @@ public static Object decodePrivateKey(Object keyStoreValue, String keyAlias, Str
             throw CryptoUtils.createError("PKCS12 key store not found at: " + keyStoreFile.getAbsoluteFile());
         } catch (KeyStoreException | CertificateException | IOException e) {
             throw CryptoUtils.createError("Unable to open keystore: " + e.getMessage());
+        } catch (NoSuchAlgorithmException e) {
+            return CryptoUtils.createError("Algorithm for key recovery is not found: " + e.getMessage());
+        } catch (NullPointerException e) {
+            return CryptoUtils.createError("Key cannot be recovered by using given keyAlias [" + keyAlias +
+                    "] and keyPassword [" + keyPassword + "]");
```
Comment (ldclakmal) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java]:
```suggestion
            return CryptoUtils.createError("Key cannot be recovered by using given key alias: [" + keyAlias + "] +
                         " and key password: [" + keyPassword + "]");
```

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java [hunk 2]
```diff
@@ -85,6 +78,13 @@ public static Object decodePrivateKey(Object keyStoreValue, String keyAlias, Str
             throw CryptoUtils.createError("PKCS12 key store not found at: " + keyStoreFile.getAbsoluteFile());
         } catch (KeyStoreException | CertificateException | IOException e) {
             throw CryptoUtils.createError("Unable to open keystore: " + e.getMessage());
+        } catch (NoSuchAlgorithmException e) {
+            return CryptoUtils.createError("Algorithm for key recovery is not found: " + e.getMessage());
+        } catch (NullPointerException e) {
+            return CryptoUtils.createError("Key cannot be recovered by using given key alias: [" + keyAlias +
+                         "] and key password: [" + keyPassword + "]");
```
Comment (anupama-pathirage) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java]: If we consider security best practices, is it ok to have the `keyPassword` in error message and logs? @ldclakmal

Comment (kalaiyarasiganeshalingam) [reply stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java]: Fixed.

Comment (ldclakmal) [reply stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java]: > If we consider security best practices, is it ok to have the `keyPassword` in error message and logs? @ldclakmal

Yes we cannot. Sorry, I missed it.

**Pull request merged**